single_file_stac_api/backend.py

```diff
@@ -0,0 +1,229 @@
+"""single_file_stac_api.backend"""
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Union
+from urllib.parse import urljoin
+
+from rtree import index
+from stac_api import config
+from stac_api.clients.base import BaseCoreClient, BaseTransactionsClient, NumType
+from stac_api.models import schemas
+from stac_api.models.links import CollectionLinks, ItemLinks
+from stac_pydantic import Collection, Item, ItemCollection
+from stac_pydantic.api import ConformanceClasses, LandingPage
+from stac_pydantic.extensions.single_file_stac import SingleFileStac
+from stac_pydantic.shared import Link, MimeTypes, Relations
+
+
+@dataclass
+class Database:
+    """cheapo spatial database.
+
+    https://rtree.readthedocs.io/en/latest/tutorial.html#using-rtree-as-a-cheapo-spatial-database
+    """
+
+    collections: List[Collection] = field(default_factory=list)
+    items: List[Item] = field(default_factory=list)
+
+    def __post_init__(self):
+        """post init handler"""
+        self.host = f"http://{config.settings.host}:{config.settings.port}"
+        self.index = index.Index()
+
+    def intersects(self, bbox):
+        """find all items which intersect the bbox."""
+        return [n.object for n in self.index.intersection(bbox, objects=True)]
```

Comment:
@geospatial-jeff because Rtree uses bbox to store in the index it might be good to filter based on the exact item geometry on a second step 🤷‍♂️ (maybe using pygeos vectorize operation for speed)